pkg/serviceaccount/metadata_test.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serviceaccount_test
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	restful "github.com/emicklei/go-restful"
+	jose "gopkg.in/square/go-jose.v2"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+)
+
+func setupServer(t *testing.T, iss string, keys []interface{}) *httptest.Server {
+	t.Helper()
+
+	c := restful.NewContainer()
+	s := httptest.NewServer(c)
+
+	// Install after we start server s.t. key URL can include host
+	srv, err := serviceaccount.NewServer(iss, s.URL, keys)
+	if err != nil {
+		t.Fatal(err)
+	}
+	srv.SetErrorHandler(func(err error) {
+		t.Fatal(err)
+	})
+	srv.Install(c)
```

Comment:
nit: Install and Set before starting the server.

Reply:
Looks like we need the port for the URL passed to serviceaccount.NewServer, which you only get after starting it.